include/deal.II/grid/tria_iterator.h

```diff
@@ -464,6 +464,16 @@ class TriaRawIterator : public std::iterator<std::bidirectional_iterator_tag,Acc
    */
   std::size_t memory_consumption () const;
 
+  /**
+   * Mark the class as bidirectional iterator and declare some typedefs which
+   * are standard for iterators and are used by algorithms to enquire about the
+   * specifics of the iterators they work on.
+   */
+  typedef std::bidirectional_iterator_tag iterator_category;
+  typedef Accessor                        value_type;
+  typedef std::ptrdiff_t                  difference_type;
```

Comment:
This could probably be an `int` -- we store the indices within a level only as `int`s.